### PR TITLE
chore: add mergify

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
We should automate some stuff on isomer, which would otherwise either slip by or require man hours. 

## Solution
Add mergify to the repo to automate the boring stuff away. The mergify config is relatively conservative (taken from `ts-template`) with a few added configurations. The added configs are: 

1. close PRs without any recent (>1 month) activity 
2. delete branches after their PR is merged (we can still restore branch so this is not information destroying) 
3. mark a PR as draft if [wip] (case insensitive) is in the title.

## Note
1. the mergify config only runs `lint` and does **not** test to see if the changes pass (for dependabot/snyk). this is due to the extremely long run time for ci tests - we should test them once we have lightweight unit tests up.

### CODE REVIEW

File: .github/mergify.yml
Comment (prestonlimlianjie) [inline .github/mergify.yml:4]: Should just be `author=dependabot[bot]`?

Dependabot preview is shut down now:
https://github.blog/2021-04-29-goodbye-dependabot-preview-hello-dependabot/

File: .github/mergify.yml
Comment (prestonlimlianjie) [inline .github/mergify.yml:24]: Let's hold off on this one for now while we clean up the PR/issues

**Review by prestonlimlianjie — CHANGES_REQUESTED** [review]

File: .github/mergify.yml
Comment (seaerchin) [reply .github/mergify.yml:6]: you're absolutely right - i got this from [ts-template](https://github.com/opengovsg/ts-template/blob/develop/.github/mergify.yml) and assumed it was correct...

we should also make a corresponding PR upstream!

File: .github/mergify.yml
Comment (seaerchin) [reply .github/mergify.yml:4]: didn't know about this - will do! (same as above, should also fix upstream)

File: .github/mergify.yml
Comment (seaerchin) [reply .github/mergify.yml:24]: unsure if you're referring to just this rule (closing stale PRs) or mergify in general. 

if it's just the rule, i can split the PR and make this rule its separate PR so that we can get mergify in first and the stale PR rule can be merged when the clean up is done!

Comment (prestonlimlianjie) [reply .github/mergify.yml:24]: Just the rule, as discussed earlier! :)

Comment (seaerchin) [reply .github/mergify.yml]: done [here](https://github.com/isomerpages/isomercms-frontend/pull/745/commits/0ce96f8ab1f05810f23a07ed4976694099257f84)

**Review by prestonlimlianjie — APPROVED** [review]
lgtm

**Pull request merged**
